ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands2.java

```diff
@@ -3359,7 +3359,58 @@ private void runStatementOnDriverWithAbort(String stmt) {
     } catch (CommandProcessorException e) {
     }
   }
+  @Test
+  public void testShowCompactWithFilterOption() throws Exception {
```

Comment:
Please also add some tests where there are multiple independent filters set up (like pool and partition name) to check if the filtering works well. You could also check the value in the result if that really matches the filtering criteria. Like if you filter for table = bar1, check it in the result.

Reply:
Added these conditions in tests